cucumber/cucumber-js
# update gherkin etc, validate against cck

More or less copied the approach from [cucumber-jvm](https://github.com/cucumber/cucumber-jvm/tree/master/compatibility).

To run:

```
$ yarn
$ yarn cck-test
```

(Or, run `compatibility/cck_spec.ts` from your IDE.)

Done:

- Expose `World` and `IWorldOptions` in public API
- Add a `log` function to the World which attaches text with the cucumber log media type
- Emit attachments in new structure
- Add some methods to `DataTable`
- Emit `Meta` message at start of run - fixes #1310 
- Keep a stopwatch of the test run, use it for a `timestamp` on various messages
- Emit messages for hooks and step definitions
- Emit messages for parameter type definitions
- `stepMatchArgumentsLists` are now being emitted within `testCase.testSteps`
- `pickleAccepted` and `pickleRejected` messages are now not emitted
- `testCaseFinished.testResult` is now not emitted (see https://github.com/cucumber/cucumber/issues/902)
- `testRunFinished.success` is now not emitted

Questions:

- `testCase` messages per CCK fixtures should be emitted together at start of run, rather than before each test case is started (non-trivial refactor - do we need this right away?)
